pkgs/development/compilers/gcc/4.8/default.nix

```diff
@@ -470,7 +472,7 @@ stdenv.mkDerivation ({
 
   LIBRARY_PATH = concatStrings
                    (intersperse ":" (map (x: x + "/lib")
-                                         (optionals (zlib != null) [ zlib ]
+                                         (optionals (zlib != null) [ zlib.out ]
```

Comment:
`concatStrings (intersperse ":" (map (x: x + "/lib")` should be able to be shortened to `lib.makeLibraryPath`, then `.out` from `zlib.out` should be able to be dropped.

(I actually wonder how this even works on current `master` since these resolve to e.g. `${gmp.dev}/lib`).